libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -93,19 +96,31 @@
     private String userHint;
     private String spActivityName;
     private Bundle spActivityExtras;
+    private BlockingQueue<Boolean> authConfigReady;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
-        // Getting login options from intent's extras
+        // Getting login options from intent's extras.
         final LoginOptions loginOptions = LoginOptions.fromBundle(getIntent().getExtras());
 
-        // Protect against screenshots
+        // Protect against screenshots.
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
                 WindowManager.LayoutParams.FLAG_SECURE);
 
-        // Setup content view
+        // Fetches auth config if required.
+        authConfigReady = new ArrayBlockingQueue<>(1);
```

Comment:
Shouldn't you move the test for production or sandbox here and not even create a AuthConfigTask if there is not auth config expected?

Reply:
We need that check in the `AsyncTask` for the `Change Server` use case (and any other use situations where we might want to fetch that value in the future). I figured it was better to centralize that code there. It should return pretty quickly in that case since there's no network call.